functional and tested setTrieFlushIntervalClone plugin

philip-morlier opened the pull request.

holiman committed:
params: begin v1.11.2 release cycle

MariusVanDerWijden committed:
eth/catalyst: send INVALID instead of INVALID_BLOCK_HASH (#26696)

This change will break one hive test, but pass another and it will be the better way going forward

rkrasiuk committed:
eth/catalyst: request too large error (#26722)

The method `GetPayloadBodiesByRangeV1` now returns "-38004: Too large request" error if the requested range is too large, according to spec

Co-authored-by: Martin Holst Swende <martin@swende.se>

holiman committed:
core/trie: remove trie tracer (#26665)

This PR contains a small portion of the full pbss PR, namely

    Remove the tracer from trie (and comitter), and instead using an accessList.
    Related changes to the Nodeset.


---------

Co-authored-by: Gary Rong <garyrong0905@gmail.com>

fjl committed:
rpc: fix unmarshaling of null result in CallContext (#26723)

The change fixes unmarshaling of JSON null results into json.RawMessage.

---------

Co-authored-by: Jason Yuan <jason.yuan@curvegrid.com>
Co-authored-by: Jason Yuan <jason.yuan869@gmail.com>

philip-morlier committed:
functional and tested setTrieFlushIntervalClone plugin

karalabe committed:
build: ship bootstrapper Go along with builder for PPA (#26731)

karalabe committed:
build: fix setting env var, temp early exit

karalabe committed:
build: fix gobootstrap path for the PPA

karalabe committed:
build: add some PPA debug logs, sigh

karalabe committed:
internal/build: revert raising the chunk size for PPA

karalabe committed:
build: yet another weird PPA fix

karalabe committed:
build: fix (finaly?) the PPA env vars for Go bootstrapping

karalabe committed:
build: fix Go 1.19.0 bootstrapper issues on 386 PPA

karalabe committed:
build: enable Lunar Lobster PPA builds

rjl493456442 committed:
Revert "core/trie: remove trie tracer (#26665)" (#26732)

This reverts commit 7c749c947a9d5181f5f2c1b3fdb5ea6b0e401e8e.

AusIV commented on core/plugin_hooks.go:
```diff
@@ -296,3 +297,24 @@ func pluginGetBlockTracer(hash common.Hash, statedb *state.StateDB) (*metaTracer
+	for _, fni := range fnList {
+		log.Error("len fn list", "len", len(fnList))
+		if fn, ok := fni.(func(time.Duration) time.Duration); ok {
+			return fn(flushInterval) 
```
```flushInterval = fn(flushInterval)```

philip-morlier committed:
added warn message to setTrieFlushIntervalClone hook

AusIV commented:
See https://github.com/openrelayxyz/plugeth-utils/pull/18#discussion_r1112262027

Also, tests are failing, seemingly to do with a null pointer dereference in your hook.

AusIV commented on core/plugin_hooks.go:
```diff
@@ -296,3 +298,27 @@ func pluginGetBlockTracer(hash common.Hash, statedb *state.StateDB) (*metaTracer
+
+func pluginSetTrieFlushIntervalClone(flushInterval time.Duration) time.Duration {
+	if plugins.DefaultPluginLoader == nil {
+		log.Warn("Attempting setTreiFlushIntervalClone, but default PluginLoader has not been initialized")
```
This should have a `return` statement after it so we don't invoke the plugin with a nil DefaultPluginLoader

swkim101 committed:
cmd/geth: clarify dumpconfig options (#26729)

Clarifies the documentation around dumpconfi

Signed-off-by: Sungwoo Kim <git@sung-woo.kim>

karalabe committed:
core, eth: merge snap-sync chain download progress logs (#26676)

holiman committed:
core: fix accessor mismatch for genesis state (#26747)

karalabe committed:
core/rawdb: expose chain freezer constructor without internals (#26748)

rjl493456442 committed:
all: use unified emptyRootHash and emptyCodeHash (#26718)

The EmptyRootHash and EmptyCodeHash are defined everywhere in the codebase, this PR replaces all of them with unified one defined in core/types package, and also defines constants for TxRoot, WithdrawalsRoot and UncleRoot

yierx committed:
eth/filters: fix a breaking change and return rpctransaction (#26757)

* eth/filters: fix a breaking change and return rpctransaction

* eth/filters: fix test cases

---------

Co-authored-by: Catror <me@catror.com>

karalabe committed:
common/math: allow HexOrDecimal to accept unquoted decimals too (#26758)

karalabe committed:
params: release Geth v1.11.2

philip-morlier committed:
Merge commit '73b01f40c' geth v1.11.2 into merge/geth-v1.11.2

philip-morlier committed:
Modified chain freezer map keys to Public to match v1.11.2

philip-morlier committed:
Added return to trie flush hook, updated to utils v0.0.23.

philip-morlier committed:
updtated utils to v0.0.24

AusIV committed:
Merge pull request #70 from openrelayxyz/merge/geth-v1.11.2

Merge/geth v1.11.2

philip-morlier committed:
Merge remote-tracking branch 'origin/develop' into feature/trie_interval_clone
